### ISSUE TITLE
Interactive stack new

### ISSUE THREAD

**Issue by mgsloan** [commercialhaskell/stack#1565]
In the past, interactive `stack new` [was discussed](https://github.com/commercialhaskell/stack/issues/253#issuecomment-110648747).  I'm opening this issue to track adding an interactive prompt for template parameters.  Leaving the prompts empty and just pressing enter should select a default value.

Additionally, an excellent suggestion comes from [wcauchois[(https://github.com/wcauchois), which is to attempt to get name / email info from git.  If this info is available, use it as a default in the interactive prompt.

**Comment by gambogi** [comment]
Unless someone else is working on this, I'll take it

**Comment by mgsloan** [comment]
Go for it!

**Comment by gambogi** [comment]
Do we only want `NewOpts` interactively? Or would it be appropriate to also get `InitOpts`?

Where should the interactive code live? It feels natural for it to go in `Stack.New`. If other commands do interaction (and it seems like there have been discussions to that effect) the actual interaction helper functions should probably be separate. A module like that doesn't already exist, right?

I was looking over [Cabal's init implementation][1], and was thinking about taking it in that direction if that seems reasonable.

[1]: https://github.com/haskell/cabal/blob/master/cabal-install/Distribution/Client/Init.hs

**Comment by mgsloan** [comment]
> Where should the interactive code live? It feels natural for it to go in `Stack.New`. If other commands do interaction (and it seems like there have been discussions to that effect) the actual interaction helper functions should probably be separate. A module like that doesn't already exist, right?

Right, a module like that doesn't exist yet.  It makes sense to me to create a module of utilities which handle user input stuff, though.  Here's the only other spot, AFAIK, where stack asks for user input: https://github.com/commercialhaskell/stack/blob/master/src/Stack/Ghci.hs#L212
